ast.rkt

```diff
@@ -130,16 +130,57 @@
      (table-identifier self))]
   #:methods gen:serializable
   [(define (serialize self #:size [_size #f])
+     (define (serialize-constraints constraint-list)
+       (define (serialize-constraint constraint)
+         (let* [(serialized-constraint (string->bytes/utf-8 (call-with-output-string
+                                                             (lambda (s-port)
+                                                               (write (syntax->datum constraint) s-port)))))
```

Comment:
I wonder if we can use `call-with-output-bytes` instead of outputting to a string and converting to bytes afterwards. I'm pretty sure the default conversion of text to bytes is UTF-8, so it should be possible, but we'd have to double-check that.